protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -1661,6 +1665,33 @@ func renderServices(services []*descriptor.Service, paths openapiPathsObject, re
 	return nil
 }
 
+// Returns the openapiPathItemObject associated with a path. If path is not present, returns
+// empty openapiPathItemObject and false.
+func getPathItemObject(paths openapiPathsObject, path string) (openapiPathItemObject, bool) {
+	for _, pathData := range paths {
+		if pathData.Path == path {
+			return pathData.PathItemObject, true
+		}
+	}
+
+	return openapiPathItemObject{}, false
+}
+
+// If a path already exists in openapiPathsObject, updates that path's openapiPathItemObject. If not,
+// appends a new path and openapiPathItemObject to the openapiPathsObject.
+func updatePaths(paths *openapiPathsObject, path string, pathItemObject openapiPathItemObject) {
+	for i, p := range *paths {
+		if p.Path == path {
+			(*paths)[i].PathItemObject = pathItemObject
+			return
+		}
+	}
+	*paths = append(*paths, pathData{
+		Path:           path,
+		PathItemObject: pathItemObject,
+	})
+}
```

Comment:
This isn't something we need to change, but it upsets me that `renderServices` has to mutate one of the input paramers 😂.

Reply:
No, as I said, this is fine. It would be nice to refactor this code in a separate PR though if you are interested. Ideally the rendering should be separate from the mutation of the inputs.